refactor: support update testType DTO

IliaSolovev opened the pull request.

Commit:
refactor: support update testType DTO

Commit:
feat: add information about test path in covered methods modal

Commit:
feat: add support custom tests type in t2r modal on SG page

Commit:
fix: add space before "covered methods" count in header

RomanAchilov commented on src/components/tests-to-run-modal/tests-to-run-modal.tsx:
```diff
@@ -13,21 +13,24 @@
 import {
   Icons, Dropdown, Modal,
-  useCloseModal,
+  useCloseModal, capitalize,
```
```suggestion
  useCloseModal,
```

RomanAchilov commented on src/components/tests-to-run-modal/tests-to-run-modal.tsx:
```diff
@@ -45,20 +48,22 @@ export const TestsToRunModal = () => {
   const { byType: testsToRun = {} } = useGroupData<GroupedTestToRun>("/group/data/tests-to-run", groupId) || {};
-  const allTests = Object.values(testsToRun).reduce((acc, tests) => [...acc, ...tests], []);
-  const [selectedFilter, setSelectedFilter] = useState("all");
+  const allTests = useMemo(() => Object.values(testsToRun).reduce((acc, value) => [...acc, ...value], []), [testsToRun]);
```
usememo with {} ?

IliaSolovev replied:
it will work because between rerender because testsToRun will the same object

RomanAchilov commented on src/components/tests-to-run-modal/tests-to-run-modal.tsx:
```diff
@@ -45,20 +48,22 @@ export const TestsToRunModal = () => {
   const closeModal = useCloseModal("/tests-to-run-modal");
 
+  const dropdownItems = Object.entries(testsToRun).map(([key, value]) => ({
+    value: key, label: `${capitalize(key)} (${value.length})`,
```
maybe it's worth fixing this component so that the items styles can be customized?

use just
tw`capitalize ` or tw`first-line:uppercase`

IliaSolovev replied:
i can add ability change classes  on dropdown but not obvious which class has changed

RomanAchilov replied:
mb need refactor
`
<Dropdown>
  arr.map => <Dropdown.Item tw="classes" />
</Dropdown>
`

RomanAchilov commented on src/components/tests-to-run-modal/tests-to-run-modal.tsx:
```diff
@@ -100,21 +105,17 @@ export const TestsToRunModal = () => {
           <div tw="text-14 mt-4 px-6 space-y-4">
-            {(getSelectedTests() || []).map((test) => (
-              <div tw="flex items-center gap-x-4" key={test.name} data-test="tests-to-run-modal:tests-list:test">
+            {(getSelectedTests() || []).map(({ id = "", details }) => (
```
```suggestion
            {getSelectedTests().map(({ id = "", details }) => (
```
check pls

RomanAchilov commented on src/components/tests-to-run-modal/tests-to-run-modal.tsx:
```diff
@@ -13,21 +13,24 @@
 import { PLUGIN_ID } from "common";
+import { TestDetails } from "types";
 import { getTestsToRunURL, TestsToRunUrl } from "../tests-to-run-url";
+import { concatName } from "../../utils/transform-tests";
```
This function has a bad name if it is used separately from the main function

RomanAchilov commented on src/modules/associated-test-modal/associated-test-modal.tsx:
```diff
@@ -95,7 +90,14 @@ export const AssociatedTestModal = () => {
             )}
             gridTemplateColumns="400px 456px auto"
-            data={tests}
+            data={tests.map((test) => ({
```
mb improve testsTransform ?

IliaSolovev replied:
so testsTransform will be very difficult

Commit:
fix: review remarks

Commit:
feat: add link to associated test from modal

Commit:
# Conflicts:
#	src/modules/associated-test-modal/associated-test-modal.tsx
#	src/modules/covered-methods-modal/covered-methods-modal.tsx

# Conflicts:
#	src/modules/associated-test-modal/associated-test-modal.tsx
#	src/modules/covered-methods-modal/covered-methods-modal.tsx

IliaSolovev commented:
already merged
